pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -178,7 +178,15 @@ public QueryExecutorImpl(PGStream pgStream,
 
   @Override
   public int getProtocolVersion() {
-    return 3;
+    return getProtocolMajorVersion();
+  }
+
+  public int getProtocolMajorVersion() {
+    return (protocol >> 16) & 0xFF;
+  }
+
+  public int getProtocolMinorVersion() {
+    return protocol & 0xFF;
   }
```

Comment:
WDYT of one of the following?

`getProtocolVersionInfo()`

* Emphasizes that this is more than just a number now.
* Common naming pattern when moving from primitive to object.

`getProtocolVersionObject()`

* Straightforward, explicitly communicates it's an object.

`getProtocolVersionDetails()`

* Suggests there might be more metadata than just a version number.

`getProtocolVersionValue()`

* Implies it's the full value, which works if the class holds major/minor etc.

`getFullProtocolVersion()`

* Indicates a more complete version compared to the basic int.

`getProtocolVersionEx()` (or `getProtocolVersionExtended()`)

* Simple, adds the "extended" idea, though a bit more API-looking.